### PR TITLE
Develop a Budget App

### PR DESCRIPTION
**Pull request by JonahKayizzi**
This pull request consists of the following;

- Set up a rails project, with rubocop and styling Linters and Gitflow
- Created a database for the Budget App according to [this](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/images/erd_diagram.png) schema with the use of rails migrations
- Created data models, relationships, and validations for all entities
- Implemented authentication using the Devise Gem
- Added controllers, methods, routes, and views for the following pages

- [ ] Splash screen

_A simple page with the name of the app and links to the sign up and log in pages._

- [ ] Sign up and log in pages

_The user should be able to register in the app with full name, email and password (all mandatory)._
_The user can log into the app using email and password._
_If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below)._

- [ ] Home page (categories page)

_When the user logs in, they are presented with the categories page._
_For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category._
_When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category._
_There is a button "add a new category" at the bottom that brings the user to the page to create a new category._

- [ ] Transactions page

_For a given category, the list of transactions is presented, ordered by the most recent._
_At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category)._
_There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction._
_When the user clicks on the "Back" button (<), the user navigates to the home page._

- [ ] Add a new category page

_The user fills out a form to create a new category, indicating their name and icon (both mandatory)._
_The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success._
_When the user clicks on the "Back" button (<), the user navigates to the home page._

- [ ] Add a new transaction page

_The user fills out a form to create a new transaction with:_
_name (mandatory)_
_amount (mandatory)_
_categories (mandatory at least one)_
_The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category._
_When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category._
- Performed unit and integration tests using Rspec
- Deployed the project online [here](https://budget-app-r5on.onrender.com) using Render.com
- Made a video demo of the project [here](https://www.loom.com/share/97c1cd7acc91468892cb806d8d11615d) using loom.

### CODE REVIEW

File: app/views/categories/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        <div class="field"> 
+            <%= form.text_field :name, class: 'form-control', placeholder: "Name" %>
+        </div>
+        <div class="field">
+            <%= form.text_field :icon, class: 'form-control', placeholder: "ICON URL" %> 
+        </div>
+        <div class="form-group form-submit">
+            <%= form.submit 'CREATE CATEGORY', class: 'btn btn-primary' %>
+        </div>
+    <% end %>
```
Comment (codepantha) [inline app/views/categories/new.html.erb:27]: - [ ] It looks like a turbo error occurs while trying to create a new category as shown below 👇 Please try to fix it
image
You can reference this [resource](https://stackoverflow.com/questions/70400958/error-form-responses-must-redirect-to-another-location) to help fix the issue

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @JonahKayizzi 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Video demo is well put together 📌 
✔️ Added tests 📌 
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/categories/new.html.erb (hunk 1 above)
Comment (JonahKayizzi) [reply app/views/categories/new.html.erb:27]: Hello @codepantha , Thanks for the review and feedback. However, I do not experience the same errors on my side. Both on the local and deployed versions of the app.

Please see [this](https://www.loom.com/share/ccf279ab65e7457e9f283db6a8c6778d) video of my chrome debugger when I add a new category

**Review by mwanawabangona — APPROVED** [review]
Hi @JonahKayizzi 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  

H
## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/categories/new.html.erb (hunk 1 above)
Comment (codepantha) [reply app/views/categories/new.html.erb:27]: Interesting! On closer inspection, I see that it only happens when the image link for the category looks like this  👇

```
data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAHwAAAB8...
```

**Pull request merged**